app/utils/callback/lava.py

```diff
@@ -0,0 +1,234 @@
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import bson
+import datetime
+import errno
+import json
+import models.boot
+import os
+import utils
+import yaml
+
+import utils.boot
+import utils.db
+import utils.lava_log_parser
+
+# copied from lava-server/lava_scheduler_app/models.py
+SUBMITTED = 0
+RUNNING = 1
+COMPLETE = 2
+INCOMPLETE = 3
+CANCELED = 4
+CANCELING = 5
+
+LAVA_JOB_RESULT = {
+    COMPLETE: 'PASS',
+    INCOMPLETE: 'FAIL',
+}
+
+def _get_definition_meta(meta, job_data):
+    """Parse the job definition meta-data from LAVA
+
+    Parse the meta-data from the LAVA v2 job definition sent with the callback
+    and populate the required fields to store in the database.
+
+    :param meta: The boot meta-data.
+    :type meta: dictionary
+    :param job_data: The JSON data from the callback.
+    :type job_data: dict
+
+    """
+    meta['board_instance'] = job_data['actual_device_id']
+
+    definition = yaml.load(job_data['definition'], Loader=yaml.CLoader)
+    meta['mach'] = definition['device_type']
+
+    meta_data_map = {
```

Comment:
I think you can move this `meta_data_map` outside of the function definition, since it's always going to be the same and is not created on the fly.

Also, I would use as its keys the values from the `models` module to avoid typos or else:
```
models.DEFCONFIG_KEY: "kernel.defconfig_base",
models.DEFCONFIG_FULL_KEY: "kernel.defconfig",
...
```